src/utils/MinimumSpanningTree.cpp

```diff
@@ -82,11 +83,13 @@ auto MinimumSpanningTree::prim(std::unordered_set<Point> vertices) const -> Adja
         //Update the distances of all points that are not in the graph.
         for (std::pair<Point*, coord_t> point_and_distance : smallest_distance)
         {
-            coord_t new_distance = vSize2(*closest_point - *point_and_distance.first);
-            if (new_distance < point_and_distance.second) //New point is closer.
+            const coord_t new_distance = vSize2(*closest_point - *point_and_distance.first);
+            const coord_t old_distance = point_and_distance.second;
+            if (new_distance < old_distance) //New point is closer.
             {
-                smallest_distance[point_and_distance.first] = new_distance;
-                smallest_distance_to[point_and_distance.first] = closest_point;
+                auto* pt =  point_and_distance.first;
+                smallest_distance[pt] = new_distance;
+                smallest_distance_to[pt] = closest_point;
```

Comment:
I don't really agree that intermediary variables like this make it any more readable. Yes, it reduces duplication, but it doesn't necessarily read more easily.